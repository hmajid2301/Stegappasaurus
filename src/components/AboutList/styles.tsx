import { StyleSheet } from "react-native";

import { fonts } from "~/common/styles";

const styles = StyleSheet.create({
  text: {
    fontFamily: fonts.bodyLight
  }
});

export default styles;
