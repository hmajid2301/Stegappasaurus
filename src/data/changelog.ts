const changelog = `
# Changelog

## [1.0.2] - 2020-02-23
Updated play store listing with new screenshots and featured graphic.
Also fixed small bugs when trying to share app and the action not being
logged correctly.

## [1.0.1] - 2020-02-01
Better error monitoring so that issues effecting user's of the app can be better tracked
and therefore resolved. It seems issues with encoding/decoding aren't currently being tracked correctly.
Can see issues in the app analytics but cannot see what is causing the errors.

## [1.0.0] - 2020-01-31
Initial Release.`;

export default changelog;
