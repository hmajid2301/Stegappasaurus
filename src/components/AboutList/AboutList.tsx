import * as React from "react";
import { Linking, Text, TouchableOpacity, View } from "react-native";
import { IconType, ListItem } from "react-native-elements";
import Rate, { AndroidMarket } from "react-native-rate";

import { ThemeColors } from "~/modules/types";
import styles from "./styles";

interface IProps {
  backgroundColor: ThemeColors;
  color: ThemeColors;
  items: IAboutItem[];
}

export interface IAboutItem {
  title: string;
  function_to_call?: "browser" | "store";
  icon: {
    color: string;
    name: string;
    type: IconType;
  };
  url?: string;
}

export default class Progress extends React.Component<IProps, {}> {
  public render() {
    return (
      <View>
        {this.props.items.map(item =>
          this.renderListItem(
            this.props.backgroundColor,
            this.props.color,
            item
          )
        )}
      </View>
    );
  }

  private renderListItem = (
    backgroundColor: ThemeColors,
    color: ThemeColors,
    item: IAboutItem
  ) => (
    <ListItem
      containerStyle={{ backgroundColor }}
      key={item.title}
      leftIcon={{
        ...item.icon,
        onPress: this.chooseFunction.bind(this, item)
      }}
      topDivider={true}
      bottomDivider={true}
      title={
        <TouchableOpacity onPress={this.chooseFunction.bind(this, item)}>
          <Text style={[styles.text, { color }]}>{item.title}</Text>
        </TouchableOpacity>
      }
    />
  );

  private async chooseFunction(item: IAboutItem) {
    switch (item.function_to_call) {
      case "store":
        const options = {
          GooglePackageName: "com.stegappasaurus",
          openAppStoreIfInAppFails: true,
          preferredAndroidMarket: AndroidMarket.Google
        };
        Rate.rate(options, () => null);
        break;

      default:
        if (item.url !== undefined) {
          await Linking.openURL(item.url);
        }
        break;
    }
  }
}
