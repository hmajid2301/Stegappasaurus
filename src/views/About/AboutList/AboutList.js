import { StoreReview, WebBrowser } from 'expo';
import { Body, Button, Icon, Left, List, ListItem } from 'native-base';
import PropTypes from 'prop-types';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

import styles from './styles';


const chooseFunction = (item) => {
  switch (item.function) {
    case 'browser':
      WebBrowser.openBrowserAsync(item.url);
      break;

    case 'store':
      StoreReview.requestReview();
      break;

    default:
      break;
  }
};

const renderListItem = (item, color) => (
  <ListItem icon key={item.title}>
    <Left>
      <Button onPress={() => chooseFunction(item)} style={{ backgroundColor: item.color }}>
        <Icon {...item.icon} />
      </Button>
    </Left>
    <Body>
      <TouchableOpacity onPress={() => chooseFunction(item)}>
        <Text style={[styles.text, { color }]}>{item.title}</Text>
      </TouchableOpacity>
    </Body>
  </ListItem>
);

const AboutList = ({ color, icons }) => (
  <List>
    {
      icons.map(item => (
        renderListItem(item, color)
      ))
    }
  </List>
);

AboutList.propTypes = {
  color: PropTypes.string.isRequired,
  icons: PropTypes.array.isRequired,
};

export default AboutList;
