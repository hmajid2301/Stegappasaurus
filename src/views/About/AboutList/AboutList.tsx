import { StoreReview, WebBrowser } from "expo";
import { Body, Button, Icon, Left, List, ListItem } from "native-base";
import React from "react";
import { Text, TouchableOpacity } from "react-native";

import styles from "./styles";

interface IProps {
  icons: Icons[];
  color: string;
}

interface Icons {
  title: string;
  color: string;
  function: string;
  icon: {
    name: string;
    type: string;
  };
  url: string;
}

const chooseFunction = (item: Icons) => {
  switch (item.function) {
    case "browser":
      WebBrowser.openBrowserAsync(item.url)
        .then()
        .catch();
      break;

    case "store":
      StoreReview.requestReview()
        .then()
        .catch();
      break;

    default:
      break;
  }
};

const renderListItem = (color: string, item: Icons) => (
  <ListItem icon key={item.title}>
    <Left>
      <Button
        onPress={() => chooseFunction(item)}
        style={{ backgroundColor: item.color }}
      >
        <Icon {...item.icon} />
      </Button>
    </Left>
    <Body>
      <TouchableOpacity onPress={() => chooseFunction(item)}>
        <Text style={[styles.text, { color }]}>{item.title}</Text>
      </TouchableOpacity>
    </Body>
  </ListItem>
);

const AboutList = ({ color, icons }: IProps) => (
  <List>{icons.map(item => renderListItem(color, item))}</List>
);

export default AboutList;
