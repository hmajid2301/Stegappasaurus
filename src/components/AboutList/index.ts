import AboutList, {AboutItem as OtherAboutItem} from './AboutList';

export type AboutItem = OtherAboutItem;
export default AboutList;
