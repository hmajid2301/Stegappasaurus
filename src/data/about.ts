import meta from '~/../app.json';
import {AboutItem} from '~/components/AboutList/AboutList';

const about = (color: any): AboutItem[] => [
  {
    icon: {
      color,
      name: 'versions',
      type: 'octicon',
    },
    title: `Version ${meta.version}`,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    icon: {
      color,
      name: 'code-fork',
      type: 'font-awesome',
    },
    title: 'Fork this Project on GitHub',
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    icon: {
      color,
      name: 'gitlab',
      type: 'font-awesome',
    },
    title: 'Personal GitLab',
    url: 'https://gitlab.com/hmajid2301',
  },
  {
    icon: {
      color,
      name: 'web',
      type: 'material-community',
    },
    title: 'Personal Website',
    url: 'https://haseebmajid.dev',
  },
  {
    icon: {
      color,
      name: 'rate-review',
      type: 'material',
    },
    title: 'Rate the app',
  },
  {
    icon: {
      color,
      name: 'paypal',
      type: 'font-awesome',
    },
    title: 'Buy me a coffee',
    url: 'https://www.paypal.me/hmajid2301',
  },
];

export default about;
