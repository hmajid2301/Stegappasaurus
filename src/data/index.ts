import about from "./about";
import questions from "./questions";
import slides from "./slides";

export { about, questions, slides };
