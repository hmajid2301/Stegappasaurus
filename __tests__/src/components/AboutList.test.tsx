import {mount, shallow} from 'enzyme';
import React from 'react';
import {Linking} from 'react-native';
import Rate from 'react-native-rate';

import AboutList from '~/components/AboutList';
import {about} from '~/data';

jest.mock('react-native-rate');

describe('AboutList: Match snapshots', () => {
  test('1', () => {
    const component = shallow(
      <AboutList items={about} color="#17212D" backgroundColor="#FFF" />,
    );
    expect(component).toMatchSnapshot();
  });

  test('2', () => {
    const component = shallow(
      <AboutList items={about} color="#FFF" backgroundColor="#17212D" />,
    );
    expect(component).toMatchSnapshot();
  });

  test('3', () => {
    const component = shallow(
      <AboutList items={[]} color="#FFF" backgroundColor="#17212D" />,
    );
    expect(component).toMatchSnapshot();
  });

  test('List Item Title', () => {
    const component = mount(
      <AboutList items={about} color="#17212D" backgroundColor="#FFF" />,
    );
    const title = (component
      .find('ListItem')
      .first()
      .props() as any).title;
    expect(title).toMatchSnapshot();
  });
});

describe('AboutList: OnPress Props', () => {
  test('ListItem leftIcon onPress: Web Browser', () => {
    const component = mount(
      <AboutList
        items={[
          {
            icon: {
              color: '#fff',
              name: 'versions',
              type: 'octicon',
            },
            title: 'Version 0.1.0',
            url: 'https://github.com/hmajid2301/Stegappasaurus',
          },
        ]}
        color="#FFF"
        backgroundColor="#17212D"
      />,
    );

    const spy = jest.spyOn(Linking, 'openURL').mockResolvedValue({
      status: true,
    });

    (component
      .find('ListItem')
      .first()
      .props() as any).leftIcon.onPress();

    expect(spy).toHaveBeenCalled();
  });

  test('TouchableOpacity onPress: Web Browser', () => {
    const component = mount(
      <AboutList
        items={[
          {
            icon: {
              color: '#fff',
              name: 'versions',
              type: 'octicon',
            },
            title: 'Version 0.1.0',
            url: 'https://github.com/hmajid2301/Stegappasaurus',
          },
        ]}
        color="#FFF"
        backgroundColor="#17212D"
      />,
    );

    const spy = jest.spyOn(Linking, 'openURL');

    (component
      .find('TouchableOpacity')
      .first()
      .props() as any).onPress();

    expect(spy).toHaveBeenCalled();
  });

  test('TouchableOpacity onPress: No URL', () => {
    const component = mount(
      <AboutList
        items={[
          {
            icon: {
              color: '#fff',
              name: 'versions',
              type: 'octicon',
            },
            title: 'Version 0.1.0',
          },
        ]}
        color="#FFF"
        backgroundColor="#17212D"
      />,
    );

    const spy = jest.spyOn(Linking, 'openURL');

    (component
      .find('TouchableOpacity')
      .first()
      .props() as any).onPress();

    expect(spy).toHaveBeenCalled();
  });

  test('TouchableOpacity onPress: Rate app', () => {
    const component = mount(
      <AboutList
        items={[
          {
            function_to_call: 'store',
            icon: {
              color: '#ff0000',
              name: 'rate-review',
              type: 'material',
            },
            title: 'Rate the App',
          },
        ]}
        color="#FFF"
        backgroundColor="#17212D"
      />,
    );

    const spy = jest.spyOn(Rate as any, 'rate');

    (component
      .find('TouchableOpacity')
      .first()
      .props() as any).onPress();
    expect(spy).toHaveBeenCalled();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
