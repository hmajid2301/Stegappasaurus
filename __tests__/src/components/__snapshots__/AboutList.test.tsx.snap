// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutList: Match snapshots 1 1`] = `
<View>
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    key="Version 1.0.0-alpha.3"
    leftIcon={
      Object {
        "color": "#009CFF",
        "name": "versions",
        "onPress": [Function],
        "type": "octicon",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#17212D",
              },
            ]
          }
        >
          Version 1.0.0-alpha.3
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    key="Fork this Project on GitHub"
    leftIcon={
      Object {
        "color": "#F50AE1",
        "name": "code-fork",
        "onPress": [Function],
        "type": "font-awesome",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#17212D",
              },
            ]
          }
        >
          Fork this Project on GitHub
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    key="Personal GitLab"
    leftIcon={
      Object {
        "color": "#FF9501",
        "name": "gitlab",
        "onPress": [Function],
        "type": "font-awesome",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#17212D",
              },
            ]
          }
        >
          Personal GitLab
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    key="Personal Website"
    leftIcon={
      Object {
        "color": "#4CDA64",
        "name": "web",
        "onPress": [Function],
        "type": "material-community",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#17212D",
              },
            ]
          }
        >
          Personal Website
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    key="Rate the App"
    leftIcon={
      Object {
        "color": "#FD3C2D",
        "name": "rate-review",
        "onPress": [Function],
        "type": "material",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#17212D",
              },
            ]
          }
        >
          Rate the App
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
</View>
`;

exports[`AboutList: Match snapshots 2 1`] = `
<View>
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    key="Version 1.0.0-alpha.3"
    leftIcon={
      Object {
        "color": "#009CFF",
        "name": "versions",
        "onPress": [Function],
        "type": "octicon",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Version 1.0.0-alpha.3
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    key="Fork this Project on GitHub"
    leftIcon={
      Object {
        "color": "#F50AE1",
        "name": "code-fork",
        "onPress": [Function],
        "type": "font-awesome",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Fork this Project on GitHub
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    key="Personal GitLab"
    leftIcon={
      Object {
        "color": "#FF9501",
        "name": "gitlab",
        "onPress": [Function],
        "type": "font-awesome",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Personal GitLab
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    key="Personal Website"
    leftIcon={
      Object {
        "color": "#4CDA64",
        "name": "web",
        "onPress": [Function],
        "type": "material-community",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Personal Website
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    key="Rate the App"
    leftIcon={
      Object {
        "color": "#FD3C2D",
        "name": "rate-review",
        "onPress": [Function],
        "type": "material",
      }
    }
    title={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Light",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Rate the App
        </Text>
      </TouchableOpacity>
    }
    topDivider={true}
  />
</View>
`;

exports[`AboutList: Match snapshots 3 1`] = `<View />`;

exports[`AboutList: Match snapshots Title 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Light",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Version 1.0.0-alpha.3
  </Text>
</TouchableOpacity>
`;
